hazelcast/src/main/java/com/hazelcast/internal/ascii/rest/HttpCommand.java

```diff
@@ -160,19 +160,80 @@ public void setResponse(@Nonnull byte[] statusLine,
         }
         response.put(TextCommandConstants.RETURN);
         response.flip();
+        setStatusCode(statusCode.code);
+    }
+
+    /**
+     * Prepares a response with the provided status line, binary-encoded
+     * content type and response.
+     * The format of response:
+     *       Status-Line (that includes CRLF itself);
+     *       for each header in header do:
+     *          header + CLRF
+     *       done;
+     *       CRLF;
+     *       response-body;
+     *
+     * @param statusCode the HTTP response status code
+     * @param headers    the map of response headers (In addition to these, we also
+     *                   add the content-length header)
+     * @param value       binary-encoded response content value
+     */
+    public void setResponseWithHeaders(HttpStatusCode statusCode,
```

Comment:
this method seems to be unused

Reply:
Yes, but it doesn't hurt either, and it doesn't hurt in terms of future backports to keep the codebase of the different supported branches as close as possible.